services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -693,7 +696,18 @@ protected void initDasCustody() {
             dataColumnSidecarValidator,
             operationPoolAsyncRunner,
             Duration.ofSeconds(1));
-    dasCustodySync = new DasCustodySync(dataColumnSidecarCustodyImpl, sidecarRetriever);
+    MiscHelpersEip7594 miscHelpersEip7594 =
+        MiscHelpersEip7594.required(spec.forMilestone(SpecMilestone.EIP7594).miscHelpers());
+    RecoveringSidecarRetriever recoveringSidecarRetriever =
+        new RecoveringSidecarRetriever(
+            sidecarRetriever,
+            kzg,
+            miscHelpersEip7594,
+            canonicalBlockResolver,
+            sidecarDB,
+            operationPoolAsyncRunner,
+            Duration.ofMinutes(5));
```

Comment:
it's a huge delay, we could stuck for almost an epoch. But probably ok for prototype

Reply:
Yep. We would definitely need some advanced logic in detecting 'lost' columns. 
With a small delay when syncing a lot of columns they could be requested by larger chunks and hit the timeout while just sitting in the `delegate` queue.